adapter-base/src/main/java/org/eclipse/hono/adapter/exception/DeviceDisabledOrNotRegisteredException.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.adapter.exception;
```

Comment:
Regarding the extra `exception` package:
In the parent `org.eclipse.hono.adapter` package, there are already some exception classes (`AdapterConnectionsExceededException`, `ConnectionDurationExceededException`, etc.).
I think it would be better to move the new exception classes to the `org.eclipse.hono.adapter` package.

Reply:
My idea is to move the other exceptions in a separate PR to the _exception_ package and I didn't do that in this PR as I tried to keep this PR short. IMHO it is better to group all of them under `exception` package.

Reply:
Grouping exceptions in an `exception` package seems rather unusual. I think they should better be put in the package that corresponds to the context in which they are usually used in or here the general `org.eclipse.hono.adapter` package.

Reply:
I have moved them to the `org.eclipse.hono.adapter` package.